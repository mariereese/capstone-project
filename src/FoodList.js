import foodData from './data/foodData.json'

export default function FoodList() {
  return (
    <ul>
      {foodData.map((food) => (
        <li>{food.food}</li>
      ))}
    </ul>
  )
}

// return foodData.map((food) => <p>{food.food}</p>)
