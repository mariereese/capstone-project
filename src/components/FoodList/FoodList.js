import styled from 'styled-components/macro'

export default function FoodList({ foodList, onAddItem }) {
  return (
    <>
      <FoodListStyled>
        {foodList.map(({ id, food }) => (
          <FoodListItem onClick={onAddItem} key={id}>
            {food}
          </FoodListItem>
        ))}
      </FoodListStyled>
    </>
  )
}

const FoodListStyled = styled.ul`
  margin: 0;
  box-shadow: 0 0 10px var(--light-grey);
  border-radius: 21px;
  width: 287px;
  max-height: 387px;
  background: white;
  padding: 0;
  overflow-y: auto;
  overflow-y: scroll;
  position: absolute;
  top: 70px;
`

const FoodListItem = styled.li`
  list-style-type: none;
  padding: 6px 10px;
  font-weight: 300;
  font-size: 1.25rem;
  color: var(--dark-grey);
`
