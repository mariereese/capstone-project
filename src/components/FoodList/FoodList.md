```js
<FoodList
  foodList={[
    {
      id: 'f1',
      food: 'Ananas, frisch',
      co2: 0.9,
    },
    {
      id: 'f2',
      food: 'Ananas, per Schiff',
      co2: 0.6,
    },
    {
      id: 'f3',
      food: 'Ananas, per Flugzeug',
      co2: 15.1,
    },
    {
      id: 'f4',
      food: 'Ananas, Dose',
      co2: 1.8,
    },
    {
      id: 'f5',
      food: 'Apfel, Durchschnitt',
      co2: 0.3,
    },
    {
      id: 'f6',
      food: 'Apfel, regional im Herbst',
      co2: 0.3,
    },
    {
      id: 'f7',
      food: 'Apfel, regional im April',
      co2: 0.4,
    },
    {
      id: 'f8',
      food: 'Apfel (Bio), Durchschnitt',
      co2: 0.2,
    },
    {
      id: 'f9',
      food: 'Apfel, aus Neuseeland',
      co2: 0.8,
    },
    {
      id: 'v10',
      food: 'Aubergine',
      co2: 0.2,
    },
    {
      id: 'f11',
      food: 'Avocado, Durchschnitt',
      co2: 0.6,
    },
    {
      id: 'f12',
      food: 'Avocado, aus Übersee',
      co2: 0.8,
    },
    {
      id: 'f13',
      food: 'Avocado (Bio), aus Übersee',
      co2: 0.8,
    },
    {
      id: 'f14',
      food: 'Banane',
      co2: 0.6,
    },
    {
      id: 'f15',
      food: 'Birne',
      co2: 0.3,
    },
    {
      id: 'v16',
      food: 'Blumenkohl',
      co2: 0.2,
    },
    {
      id: 'v17',
      food: 'Bohnen, frisch',
      co2: 0.8,
    },
    {
      id: 'v18',
      food: 'Bohnen, Dose',
      co2: 1.3,
    },
    {
      id: 'v19',
      food: 'Brokkoli, frisch',
      co2: 0.3,
    },
    {
      id: 'v20',
      food: 'Brokkoli, gefroren',
      co2: 0.7,
    },
    {
      id: 'v21',
      food: 'Champignons, frisch hell oder dunkel',
      co2: 1.3,
    },
    {
      id: 'v22',
      food: 'Champignons, Dose',
      co2: 2.4,
    },
    {
      id: 'v23',
      food: 'Erbsen, frisch grün in Schoten',
      co2: 0.4,
    },
    {
      id: 'v24',
      food: 'Erbsen, getrocknet',
      co2: 2.3,
    },
    {
      id: 'v25',
      food: 'Erbsen, gefroren',
      co2: 1.2,
    },
    {
      id: 'v26',
      food: 'Erbsen, grün Dose',
      co2: 1.7,
    },
    {
      id: 'v27',
      food: 'Erbsen, grün Glas',
      co2: 1.7,
    },
    {
      id: 'f28',
      food: 'Erdbeeren, frisch Durchschnitt',
      co2: 0.3,
    },
    {
      id: 'f29',
      food: 'Erdbeeren, frisch regional saisonal',
      co2: 0.3,
    },
    {
      id: 'f30',
      food: 'Erdbeeren, frisch aus Spanien',
      co2: 0.4,
    },
    {
      id: 'f31',
      food: 'Erdbeeren, gefroren',
      co2: 0.7,
    },
    {
      id: 'f32',
      food: 'Erdbeeren, frisch (Winter-Erdbeeren)',
      co2: 3.4,
    },
  ]}
/>
```
