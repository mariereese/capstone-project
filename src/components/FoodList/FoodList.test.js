import FoodList from './FoodList'
import { render } from '@testing-library/react'

describe('FoodList', () => {
  const testFunction = jest.fn()
  it('renders correctly', () => {
    const { container } = render(
      <FoodList
        foodList={[
          {
            id: 'f1',
            food: 'Ananas, frisch',
          },
          {
            id: 'f2',
            food: 'Ananas, per Schiff',
          },
        ]}
        onAddFood={testFunction}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
