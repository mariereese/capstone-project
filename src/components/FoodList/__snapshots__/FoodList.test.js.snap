// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodList renders correctly 1`] = `
<ul
  class="FoodList__FoodListStyled-sc-17fj5qi-0 wSPvz"
>
  <li
    class="FoodList__FoodListItem-sc-17fj5qi-1 efPrRC"
  >
    Ananas, frisch
  </li>
  <li
    class="FoodList__FoodListItem-sc-17fj5qi-1 efPrRC"
  >
    Ananas, per Schiff
  </li>
</ul>
`;
